### PR TITLE
Add an in memory attribute related repository

### PR DESCRIPTION
**Pull request by damien-carcel**
## Description

- Attribute repository
- Attribute option repository
- Attribute group repository

For each are implemented `getIdentifierProperties`, `findOneByIdentifier`, `findBy`, and `save` methods.
Every implemented methods are unit tested.

## Definition Of Done

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | OK
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: tests/back/Acceptance/Resources/config/pim/repositories.yml [hunk 1]
```diff
@@ -3,3 +3,4 @@ parameters:
     pim_catalog.repository.channel.class: Akeneo\Test\Acceptance\Channel\InMemoryChannelRepository
     pim_catalog.repository.category.class: Akeneo\Test\Acceptance\Category\InMemoryCategoryRepository
     pim_catalog.repository.currency.class: Akeneo\Test\Acceptance\Currency\InMemoryCurrencyRepository
+    pim_catalog.repository.attribute_option.class: Akeneo\Test\Acceptance\AttributeOption\InMemoryAttributeOptionRepository
```
Comment (skeleton) [inline tests/back/Acceptance/Resources/config/pim/repositories.yml]: missing InMemoryAttributeRepository

File: tests/back/Acceptance/spec/Attribute/InMemoryAttributeRepositorySpec.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace spec\Akeneo\Test\Acceptance\Attribute;
+
+use PhpSpec\ObjectBehavior;
+use Pim\Bundle\CatalogBundle\Entity\Attribute;
+use Pim\Component\Catalog\Model\AttributeInterface;
+
+class InMemoryAttributeRepositorySpec extends ObjectBehavior
+{
+    function it_returns_an_identifier_property()
```
Comment (SamirBoulil) [inline tests/back/Acceptance/spec/Attribute/InMemoryAttributeRepositorySpec.php]: We added the "it_is_a_attribute_repository_interface" as well as "it_is_a_saver_interface".

File: tests/back/Acceptance/spec/Attribute/InMemoryAttributeRepositorySpec.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    function it_finds_attributes_by_criteria()
+    {
+        $attribute = $this->createAttribute('attribute_1');
+        $this->beConstructedWith([$attribute->getCode() => $attribute]);
+
+        $this->findBy(['code' => 'attribute_1'])->shouldReturn([$attribute]);
+    }
+
```
Comment (SamirBoulil) [inline tests/back/Acceptance/spec/Attribute/InMemoryAttributeRepositorySpec.php]: What about the case where it does not find ?

File: tests/back/Acceptance/spec/Attribute/InMemoryAttributeRepositorySpec.php (hunk 2 above)
Comment (jjanvier) [reply tests/back/Acceptance/spec/Attribute/InMemoryAttributeRepositorySpec.php]: we did too

Comment (damien-carcel) [reply tests/back/Acceptance/spec/Attribute/InMemoryAttributeRepositorySpec.php]: I can add it if you really want, but Matthias aggrees it adds no value to the unit test. And it is what we are supposed to write here: unit tests, even if we are using a spec tool do to it.

We tests the behavior of the method `save`, so from the test point of view, we don't care it implements the interface or not.

File: tests/back/Acceptance/spec/Attribute/InMemoryAttributeRepositorySpec.php (hunk 3 above)
Comment (damien-carcel) [reply tests/back/Acceptance/spec/Attribute/InMemoryAttributeRepositorySpec.php]: Added.

**Pull request merged**